tests/genesis/genesis_test.go

```diff
@@ -437,6 +449,12 @@ func validateNodes(t *testing.T, genesis *registryAPI.Genesis, source *oasis.Con
 		)
 		assert.Nil(t, err)
 
+		if (&node.Node{Expiration: n.Expiration}).IsExpired(uint64(epoch)) {
+			// The indexer DB stores some nodes that are expired because
+			// an expiration event was never produced for them. Ignore them.
```

Comment:
> an expiration event was never produced for them. Ignore them.

Yikes.

Reply:
To be fair, I think those nodes' expirations are in the deep-ish past, before damask. (Not 100% sure because expiration is expressed in epochs, and I didn't check the epoch of damask genesis.) They might be special in some regard ...
I didn't investigate further, but it didn't make me think "oasis-node is randomly neglecting to emit or expose events".